packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts

```diff
@@ -28,6 +29,26 @@ export const coinjoinMiddleware =
     (api: MiddlewareAPI<Dispatch, AppState>) =>
     (next: Dispatch) =>
     (action: Action): Action => {
+        const checkIsCoinjoinResumeBlocked = (state: AppState) => {
+            const deviceStatus = selectDeviceState(state);
+            const isDeviceDisconnected = deviceStatus !== 'connected';
+            const isAccountOutOfSync = getIsCoinjoinOutOfSync(state.wallet.selectedAccount);
+            const isCoinJoinBlockedByTor = selectIsCoinjoinBlockedByTor(state);
+            const isCoinJoinDisabledByFeatureFlag = selectIsFeatureDisabled(
+                state,
+                Feature.coinjoin,
+            );
+            const isCoinjoinBlockedByRoute = state.router.route?.name === 'wallet-send';
+            return (
+                isAccountOutOfSync ||
+                isDeviceDisconnected ||
+                isCoinJoinBlockedByTor ||
+                isCoinJoinDisabledByFeatureFlag ||
+                isCoinjoinBlockedByRoute ||
+                !state.suite.online
+            );
+        };
```

Comment:
It looks lo me that this functions tries to make the same thing that `isCoinjoinResumeBlocked` from useBlockedCoinjoinResume https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/hooks/coinjoin/useBlockedCoinjoinResume.ts#L40

It would be nice to extend the some how the function in that hook with the new requirements

Reply:
Already working on that: #7524